lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -831,10 +834,19 @@ class DxilPSVWriter : public DxilPartWriter {
     PSVRuntimeInfo0 *pInfo = m_PSV.GetPSVRuntimeInfo0();
     PSVRuntimeInfo1 *pInfo1 = m_PSV.GetPSVRuntimeInfo1();
     PSVRuntimeInfo2 *pInfo2 = m_PSV.GetPSVRuntimeInfo2();
+    PSVRuntimeInfo3 *pInfo3 = m_PSV.GetPSVRuntimeInfo3();
     const ShaderModel *SM = m_Module.GetShaderModel();
     pInfo->MinimumExpectedWaveLaneCount = 0;
     pInfo->MaximumExpectedWaveLaneCount = (UINT)-1;
 
+    if (pInfo3) {
+      pInfo3->EntryFunctionName = (uint32_t)m_StringBuffer.size();
```

Comment:
At what point does the string table get set so that this uint32 that is being assigned here will correspond to the actual entry function name? Also, is it possible that the key value pair in the string table could be overwritten by another string of the same length? I don't understand how we can reliably get the entry point name from the string table at this point. It seems like m_Module has already built a string table that contains the a set of key (uint32_t) value (char*) pairs, but the key is just the length of the string, and if there are multiple entries, that seems very risky.

Reply:
> At what point does the string table get set so that this uint32 that is being assigned here will correspond to the actual entry function name?

Right here.  The current size of the string buffer is the location we are about to place this string in the code below this line. This is exactly the same pattern used for the semantic names here:

https://github.com/microsoft/DirectXShaderCompiler/blob/05094636736eaf10f964545a6873d4b31c06d739/lib/DxilContainer/DxilContainerAssembler.cpp#L697

> is it possible that the key value pair in the string table could be overwritten by another string of the same length

What key value pair?  I think you might be misinterpreting the code.  There are no key value pairs in this code.  The string table is just a `SmallVector<char, 256>` that will be written as-is to the container.  Offsets into the string table are byte offsets into this buffer.  A reader must identify where the string table is in the PSV0 container and offset into that location when looking at the offset written to the EntryFunctionName field.

https://github.com/microsoft/DirectXShaderCompiler/blob/05094636736eaf10f964545a6873d4b31c06d739/lib/DxilContainer/DxilContainerAssembler.cpp#L686

A string table could be like: `"\0FirstString\0Second\0"`.  The value written for `"FirstString"` is the offset `1`.  The value written for the `"Second"` is the offset `13`.

Here is how the offset is looked up in the string table by the reader:
https://github.com/microsoft/DirectXShaderCompiler/blob/05094636736eaf10f964545a6873d4b31c06d739/include/dxc/DxilContainer/DxilPipelineStateValidation.h#L215

Reply:
Understood, the confusion came from my assumption that GetEntryFunctionName referred to the function you defined earlier under the DxilPipelineStateValidation class, rather than the DxilModule class.
But I still don't see how the string table gets filled with the entry function name, only the string buffer. It looks like DxilPipelineStateValidation::ReadOrWrite is what initializes it, so I suppose it will have already been set before we get to this point.

Reply:
The "string table" *is* just the string buffer.  The word "table" might be misleading you here.  The string buffer gets written to the PSV0 part, and offset values in fields referring to it work by offsetting into that string buffer in the PSV0 part.

Reply:
Ah-ha, you actually identified a bug!  I was going to walk through the steps, and realized that the string buffer must be complete before we compute the size in the constructor.  We can't add to it in the write() method, because the buffer we write will still not 

I was thinking "how could the test pass?" but now I realize that it wasn't running in this branch yet due to the validator version requirement (it should now that staging has merged to main, bumping the validator version to 1.8).

I'll fix this and update the PR.

Thanks!